src/error/Makefile.am

```diff
@@ -0,0 +1,31 @@
+## Copyright (C) 1996-2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+include $(top_srcdir)/src/Common.am
+include $(top_srcdir)/src/TestHeaders.am
+
+categories.cc: forward.h $(top_srcdir)/src/mk-string-arrays.awk
+	$(AWK) -f $(top_srcdir)/src/mk-string-arrays.awk < $(srcdir)/forward.h > $@ || ($(RM) -f $@ && exit 1)
```

Comment:
```suggestion
	$(AWK) -f $(top_srcdir)/src/mk-string-arrays.awk ipath=error/ < $(srcdir)/forward.h > $@ || ($(RM) -f $@ && exit 1)
```

Reply:
Does not fix the issue.
Please check my commit 424cb777ae, it it is an acceptable solution. 
Commit comment:
> Previews "Remove the uneeded err_detail_type.h ..." commit changed
    mk-string-arrays.awk to add an include file with array definition
    only when this array is an array of SBufs. The reasons are:
      a) I (wrongly?) believed that only in the case of SBus is needed.
      b) It computed the include file name using the enum name. In the case of
         error/categories.cc the include file (error/forward.h) does not match
         the enum name (err_type)
 > This patch fixes mk-string-arrays.awk to accept an include file name
    as parameter and add it to #include files list if given

Reply:
It is not.  The AWK logic should be:

 * when enum is an array of SBuf strings SBuf.h must be included by the generated file.
> 	if (sbuf) print "#include \"sbuf/SBuf.h\""

 * when ifile is provided include the given file
>	if (ifile != "") print "#include \"" ifile "\""

 * otherwise just include the header the enum is defined in
>	else print "#include \"" nspath type ".h\""


Also, you are using the ifile parameter only for http/MethodTypes.h when this PR is needing it for error/Details.h

Reply:
I am ok with your suggestion, I fix it in commit aa65598e7.

However still I prefer my form:
 - I think make more sense a tool command line parameter which has a strict job. Eg "add this file in include files list".
 - Only the generated http/MethodType.cc needs to include the http/MethodType.h file
 - The ifile can be extended to pass more than one files to include:  `mk-string-arrays.awk ifile="sbuf/SBuf.h;http/MethodType.h" < $(srcdir)/MethodType.h`. It will cover more future cases.